doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1671 @@
+
+The IETF draft [pref-df](https://datatracker.ietf.org/doc/html/draft-ietf-bess-evpn-pref-df-13) extends the DF election algorithm assignment as below:
+- Algo 2 : Highest-Preference algorithm
+- Algo TBD: Lowest-Preference algorithm
+
+FRR 8.2 implements the highest preference based DF election algorithm. SONiC uses the same DF election algorithm.
+
+DF election will be performed in FRR and **EVPN_DF_TABLE** will be populated in APP-DB by Fpmsyncd only if the local switch is the elected DF. If EVPN_DF_TABLE entry is not present, the switch will not forward any BUM traffic to the EVPN MH Ethernet segment.
+
+**Note:** DF election timer value is set to 3 sec as per [RFC 7432](https://datatracker.ietf.org/doc/html/rfc7432), and is not configurable.
+
+**Note:** As per the standard, if two routers do not agree on a given DF election algorithm, they should fallback on default modulo based scheme proposed in [RFC 7432](https://datatracker.ietf.org/doc/html/rfc7432). This fallback option is not supported and SONiC device does not own the DF role in such interop cases.
+
+### 2.2.2.2 Interoperability between ES Multihoming VTEPs
+
+SONIC solution is fully interopable with other device supporting EVPN MH standards.
+Other device must support highest preference based DF election algorithm.
+
+It is recommended that all of the VTEPs multihoming one or more common ES are SONiC devices.
+
+<a id="223-Split-Horizon-Filtering"></a>
+### 2.2.3 Split Horizon Filtering
+Split Horizon filtering is achieved by identifying the set of VTEPs Multihoming in a given Ethernet Segment. 
+
+Using EVPN Type-4 route, FRR constructs list of participating VTEPs for each ESI. And Fpmsyncd installs EVPN_SPLIT_HORIZON_TABLE entry in APP-DB that contains list of remote VTEPs multihoming the given access interface.
+
+The Isolation group SAI object used for achieving Local Bias and Split Horizon Filtering in the MCLAG will be reused for EVPN MH. In EVPN MH an Isolation group for every Peer Multihoming VTEP is created as opposed to a single group for MCLAG. The Isolation group members are populated based on the EVPN_SPLIT_HORIZON_TABLE APP-DB entry. Traffic received from the Tunnel towards the peer multihoming VTEP will not be forwarded to the members of the Isolation group.
+
+<a id="224-Linux-Kernel-Support"></a>
+### 2.2.4 Linux Kernel Support
+EVPN Multihoming feature requires L2 next-hop group (NHG) support in the Linux kernel that is available in v5.10.
+EVPN Multihoming feature will not be supported in SONiC releases running 4.x kernel versions.
+
```

Comment:
According to [RFC 8365, Section 8.3.3](https://datatracker.ietf.org/doc/html/rfc8365#section-8.3.3)
```
VXLAN-GPE encapsulation needs to be used between these PEs and the ingress PE needs to set the BUM Traffic Bit (B bit) to indicate that this is an ingress-replicated BUM traffic.
```
It might need an additional patch to support this in the v5.10??

Reply:
@AntonButenkoGL @pbrisset Does the kernel v5.10 need patch for this?

Reply:
SONiC is currently at Linux kernel v6.1.38.

Reply:
Is the BUM Traffic Bit already supported by Linux Kernel v6.1.38?

Reply:
@gord1306 , this hld is for vxlan (RFC7432). Vxlan-gpe is still a draft (draft-ietf-nvo3-vxlan-gpe.) This hld can be extended in future to add vxlan-gpe support. Bum bit is not present in the non-gpe vxlan header. I hope this clarifies.